src/Textual/encodings/textpreprocessing.jl

```diff
@@ -0,0 +1,124 @@
+"""
+    TextEncoding() <: Encoding
+
+Encodes `Paragraph`s by applying various textual transforms.
+
+
+Encodes
+- `Paragraph` -> `Paragraph`
+
+"""
+struct Sanitize <: Encoding
+    tfms
+end
+
+function Sanitize()
+    base_tfms = [
+        replace_all_caps,
+        replace_sentence_case,
+        convert_lowercase,
+        remove_punctuations,
+        basic_preprocessing,
+        remove_extraspaces,
+    ]
+    return Sanitize(base_tfms)
+end
+
+function encodedblock(p::Sanitize, block::Paragraph)
+    return block
+end
+
+function encode(p::Sanitize, context, block::Paragraph, obs)
+    for tfm in values(p.tfms)
+        obs = tfm(obs)
+    end
+    obs
+end
+
+struct Tokenize <: Encoding
+    tfms
+end
+
+function Tokenize()
+    base_tfms = [
+        tokenize,
+    ]
+    return Tokenize(base_tfms)
+end
+
+function encodedblock(p::Tokenize, block::Paragraph)
+    return TokenVector()
+end
+
+function encode(p::Tokenize, context, block::Paragraph, obs)
+    for tfm in values(p.tfms)
+        obs = tfm(obs)
+    end
+    obs
+end
+
+function computevocabulary(data)
+    lookup_table = Dict()
+
+    enc1 = Sanitize()
+    sanitized_Data = mapobs((i) -> encode(enc1, Training(), Paragraph(), getobs(data, i)[1]), 1:25000)
```

Comment:
```suggestion
    sanitized_Data = map(i -> encode(enc1, Training(), Paragraph(), getobs(data, i)[1]), 1:25000)
```
This upper bound is a little odd to me. Why is it fixed at 250k?

Reply:
That is the total number of samples in `data`. I should have used `numobs(data)` instead. I'll change this now.

Reply:
Can you `mapobs` over `data` directly then?

Reply:
I didn't get that. Can you please elaborate?

Reply:
Instead of mapping over the indices of data and calling `getobs` for each index, why not just call `mapobs` with data?